### PR TITLE
Hide [ShortcutsModal] on [QuickopenModal] open

### LINKED ISSUE #5261: [QuickOpen] Cmd+P while Help panel is displayed should hide the help panel
**Steps to reproduce:**
1. Open the Help Panel (Click the `?`)
2. When opened, do <kbd>Cmd</kbd> + <kbd>P</kbd>

**Expected results:**

I can search for files

**Actual results:**

The help panel is in front of the search file modal

jan-31-2018 09-27-28

---

I think this can be a common behavior since you look at the help to get the shortcut and want to use it right away.

### PR DESCRIPTION
**Pull request by kootoopas**
Fixes #5261.

### Test Plan

- [x] Clicking "?" opens shortcuts
- [x] Command-P opens quickopen and closes shortcuts

### Screenshots/Videos

quickopen-open-closes-shortcuts

### CODE REVIEW

**Comment by wldcordeiro** [comment]
This would be a scenario where I would just suggest making a thunk and closing the other modal in it for both.

File: src/components/QuickOpenModal.js [hunk 1]
```diff
@@ -318,19 +327,21 @@ export class QuickOpenModal extends Component<Props, State> {
...
+
   return {
     enabled: getQuickOpenEnabled(state),
     sources: formatSources(getSources(state)),
     selectedSource,
     symbols: formatSymbols(symbols),
     query: getQuickOpenQuery(state),
-    searchType: getQuickOpenType(state)
+    searchType: getQuickOpenType(state),
+    ...getShortcutsModal(ownProps)
```
Comment (jasonLaster) [inline src/components/QuickOpenModal.js]: hmm, i'm a bit confused by this?

File: src/selectors/getShortcutsModal.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at <http://mozilla.org/MPL/2.0/>. */
+
+export default function getShortcutsModal(ownProps) {
+  return {
+    shortcutsModalEnabled: ownProps.shortcutsModalEnabled,
+    toggleShortcutsModal: ownProps.toggleShortcutsModal
```
Comment (jasonLaster) [inline src/selectors/getShortcutsModal.js]: seems like this isn't doing anything :)

File: src/reducers/sources.js [hunk 3]
```diff
@@ -174,7 +187,12 @@ function updateSource(state: Record<SourcesState>, source: Object | Source) {
     return state;
   }
 
-  return state.mergeIn(["sources", source.id], source);
+  const existingSource = state.getIn(["sources", source.id]);
+  if (existingSource) {
+    const updatedSource = existingSource.merge(source);
+    return state.setIn(["sources", source.id], updatedSource);
+  }
+  return state.mergeIn(["sources", source.id], new SourceRecordClass(source));
```
Comment (jasonLaster) [inline src/reducers/sources.js]: this is nice, but seems unreleated :)

File: src/components/QuickOpenModal.js (hunk 1 above)
Comment (kootoopas) [reply src/components/QuickOpenModal.js]: Oh gosh. I accidentally pushed an irrelevant commit from another branch too. (`Add immutable records for sources`)

File: src/reducers/sources.js (hunk 3 above)
Comment (kootoopas) [reply src/reducers/sources.js]: ditto for this one

**Comment by kootoopas** [comment]
@jasonLaster Should I open another pr? My branch is a bit messy. :)

**Comment by jasonLaster** [comment]
no worries - i think we can touch up this PR and land it

**Comment by darkwing** [comment]
Updated the PR per comments; will wait until CI passes to merge.

Thank you so much for your contribution @kootoopas !~

File: src/selectors/getShortcutsModal.js [hunk 4]
```diff
@@ -0,0 +1,9 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at <http://mozilla.org/MPL/2.0/>. */
+
+export default function getShortcutsModal(ownProps) {
```
Comment (jasonLaster) [inline src/selectors/getShortcutsModal.js]: hmm, why do we need this?

Comment (kootoopas) [reply src/selectors/getShortcutsModal.js]: i agree that we don't need it, since it maps props to... props

**Comment by kootoopas** [comment]
@darkwing Thanks! It's pretty dope to gradually get familiar with this codebase.

**Pull request merged**